### PR TITLE
#2572119 Changing execute to doExecute in VariableAdd.

### PR DESCRIPTION
**Pull request by emclaughlin1215**
See d.o. issue https://www.drupal.org/node/2572119

### CODE REVIEW

**Comment by emclaughlin1215** [comment]
Test are failing because of a protected method in the core Context.php class (setContextValue). I'm not sure what to do with that information.

**Comment by klausi** [comment]
yep, we are currently working at fixing the tests in #258

**Comment by torenware** [comment]
Pulled @emclaughlin1215's patch and ran it against today's Rules + Core with @klausi's Context patch.  The tests run green by me.

File: src/Plugin/RulesAction/VariableAdd.php [hunk 1]
```diff
@@ -29,9 +29,12 @@
 class VariableAdd extends RulesActionBase {
 
   /**
-   * Executes the plugin.
+   * Add a variable.
+   *
+   * @param string $value
```
Comment (klausi) [inline src/Plugin/RulesAction/VariableAdd.php]: The value can be of any type, so the data type should be "mixed" here.

**Comment by emclaughlin1215** [comment]
Addressed the comments and pushed up changes.

File: src/Plugin/RulesAction/VariableAdd.php [hunk 2]
```diff
@@ -29,9 +29,12 @@
...
   /**
-   * Executes the plugin.
+   * Add a variable.
+   *
+   * @param mixed $value
+   *   The variable to add.
    */
-  public function execute() {
-    $this->setProvidedValue('variable_added', $this->getContext('value')->getContextValue());
+  public function doExecute($value) {
```
Comment (klausi) [inline src/Plugin/RulesAction/VariableAdd.php:37]: should be protected, not public.

**Comment by ndf** [comment]
Moved to https://github.com/fago/rules/pull/305

**Comment by klausi** [comment]
closing

**Pull request closed without merging**